src/rules/pyupgrade/rules/extraneous_parentheses.rs

```diff
@@ -0,0 +1,143 @@
+use rustpython_parser::lexer::{LexResult, Tok};
+
+use crate::ast::types::Range;
+use crate::fix::Fix;
+use crate::registry::{Diagnostic, Rule};
+use crate::settings::{flags, Settings};
+use crate::source_code::Locator;
+use crate::violations;
+
+// See: https://github.com/asottile/pyupgrade/blob/97ed6fb3cf2e650d4f762ba231c3f04c41797710/pyupgrade/_main.py#L148
+fn match_extraneous_parentheses(tokens: &[LexResult], mut i: usize) -> Option<(usize, usize)> {
```

Comment:
@colin99d - I decided to instead do a straight port of the pyupgrade implementation based on the token stream. I didn't want to ask you to make big changes like this since you've already put in some work on these PRs. I hope that's cool though -- sorry if I should've asked first.